pages/returning-members.js

```diff
@@ -0,0 +1,282 @@
+    'Not active on Waterloop this term',
+];
+
+const CONTINUINGSTATUS = [
+    'Continue with my sub-team', 
+    'Transfer to another sub-team (please specify)', 
+    'Want to take on a leadership role - lead', 
+    'Want to take on a leadership role - co-op supervisor', 
+    'Want to become a co-op', 
+    'I\'m undecided or not continuing',
+]
+
+const FormHeader = ({title, marginBottom}) => {
+    const theme = useContext(ThemeContext);
+    return (
+        <SystemComponent
+            fontSize={theme.fontSizes.header3}
+            textAlign='center'
+            mb={marginBottom}
+        >
+            <SystemSpan>
+                <Subtitle>{title}</Subtitle>
+            </SystemSpan>
+        </SystemComponent>
+    );
+}
+
+const RegistrationForm = () => {
+    const theme = useContext(ThemeContext);
+
+    const [formValues, setFormValues] = useState({
+        fullName:  "",
+        termDescription: "", 
+        previousTerms: [], 
+        futureTerms: [], 
+        subteam: "",
+        activeStatus: "", 
+        continuingStatus: "", 
+        email: "", 
+        comments: "",
+        futureTasks: "", 
+    });
+
+    const [hasError, setHasError] = useState({
+        fullName: false,
+        termDescription: false, 
+        membership: false, 
+        previousTerms: false, 
+        futureTerms: false,
+        activeStatus: false, 
+        subteam: false,
+        activeStatus: false,
+        continuingStatus: false,
+        email: false, 
+    });
+
+    const handleSave = () => {
+        console.log('clicked button')
+        console.log(formValues)
+        const updatedErrorList = {...hasError};
+
+        for (const [name, value] of Object.entries(updatedErrorList)) {
+            updatedErrorList[name] = false;
+        }
+
+        if (!formValues.fullName || formValues.fullName.split(' ').length !== 2) {
+            updatedErrorList['fullName'] = true;
+        }
+        if (!formValues.email || !isEmail(formValues.email)) {
+            updatedErrorList['email'] = true;
+        }
+        if (!formValues.termDescription) {
+            updatedErrorList['termDescription'] = true;
+        }
+        if (!formValues.subteam) {
+            updatedErrorList['subteam'] = true;
+        }
+        if (!formValues.continuingStatus) {
+            updatedErrorList['continuingStatus'] = true;
+        }
+        if (!formValues.activeStatus) {
+            updatedErrorList['activeStatus'] = true;
+        }
+        if (!formValues.futureTerms) {
+            updatedErrorList['futureTerms'] = true;
+        }
+
+        setHasError(updatedErrorList);
+
+        if (Object.values(updatedErrorList).some(hasError => hasError === true)) {
+            return;
+        }
+        // call onSubmit API route here
+    }
+
+    const handleInputChange = (name, value) => {
+        
+        if (name === "phoneNumber") {
+            if (value && (!value.match(/^[0-9]*$/) || value.length > 10)) return;
```

Comment:
Minor: I would surround the `return;` in brackets and move it to a separate line.